aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -29,6 +29,13 @@ static inline Tensor to_impl(const Tensor& self, const TensorOptions& options, b
     return self;
   }
 
+  if (options.device().type() == DeviceType::Vulkan
```

Comment:
why exactly do you need it? to auto-contiguous the tensors? I wonder whether it's better to error out instead and fix it at higher levels of API.

At the very least - add a comment here.

cc @ezyang fyi

Reply:
Yeah, please don't do this.

Reply:
I have prepared fix in separate PR: https://github.com/pytorch/pytorch/pull/39019 (introduced empty_strided_vulkan and after that these 'ifs' can be removed, I added them before decision to use Strided layout in Vulkan and have not cleaned them after switch )